.github/workflows/gradle.yml

```diff
@@ -0,0 +1,40 @@
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
```

Comment:
Should we consider using 17 here?

Reply:
we can use a matrix to ensure it works in several versions - which versions should we test against?

Reply:
test/build fails currently in 17 with:
```General error during semantic analysis: Unsupported class file major version 61```

Reply:
gotta be gradle dependencies, right?

Reply:
Bump Gradle in `gradle/wrapper/gradle-wrapper.properties` to point to 7.4.2 or higher - working for me locally on Java 17 after that change.